File: day2/part1/day2.go
package main

import (
	"fmt"
	"io/ioutil"
	"strings"
)

func main() {
	text, err := ioutil.ReadFile("input")
	if err != nil {
		panic(err)
	}
	var ids = strings.Split(string(text), "\n")

	var twice int
	var three int
	for _, id := range ids {
		var letterFrequencies = countLetters(id)
		var isTwice = false
		var isThree = false
		for _, frequency := range letterFrequencies {
			if frequency == 2 {
				isTwice = true
			} else if frequency == 3 {
				isThree = true
			}
		}
		if isTwice {
			twice++
		}
		if isThree {
			three++
		}
	}
	var checksum = twice * three
	fmt.Printf("checksum: %v\n", checksum)
}

func countLetters(word string) map[rune]int {
	var res = make(map[rune]int)
	for _, char := range word {
		res[char]++
	}
	for k, v := range res {
		fmt.Printf("%v = %v\n", k, v)
	}
	return res
}

File: day2/part1/day2.py
#!/usr/bin/env python3

import collections
import typing

with open("input", "r") as file:
    ids: typing.List[str] = file.read().split("\n")

twice = 0
three = 0

for _id in ids:
    counter = collections.Counter(_id)
    is_twice = False
    is_three = False
    for _, count in counter.items():
        if(count == 2):
            is_twice = True
        elif(count == 3):
            is_three = True
    if(is_twice):
        twice += 1
    if(is_three):
        three += 1

checksum = twice * three

print("checksum: {}".format(checksum))
